test/cql-pytest/test_null.py

```diff
@@ -65,6 +79,51 @@ def test_insert_null_key_lwt(cql, table1):
     with pytest.raises(InvalidRequest, match='null value'):
         cql.execute(stmt, [None, s])
 
+# INSERT statements must specify all components of the primary key,
+# and the corresponding query parameters, if any, must not be null.
+# If INSERT or DELETE statement is missing a non-last element of
+# the primary key, the error message generated contained
+# an invalid column name (#12046).
+# The problem occurred if the query contains a column with the list type,
+# otherwise statement_restrictions::process_clustering_columns_restrictions
+# checks that all the components of the key are specified.
+def test_insert_with_list_column_and_missing_clustering_key_part(cql, table2):
+    key = unique_key_string()
+    with pytest.raises(InvalidRequest,
+                       match='Missing mandatory PRIMARY KEY part a|Some clustering keys are missing: a'):
+        cql.execute(cql.prepare(f"INSERT INTO {table2} (id,b,c) VALUES ('{key}',1,null)"))
+    with pytest.raises(InvalidRequest,
+                       match='Missing mandatory PRIMARY KEY part b|Some clustering keys are missing: b'):
+        cql.execute(cql.prepare(f"INSERT INTO {table2} (id,a,c) VALUES ('{key}',1,null)"))
+
+# The same as test_insert_with_list_column_and_missing_clustering_key_part but for
+# different code path in modification_statement::process_where_clause.
+# We need DELETE here to have require_full_clustering_key() == false.
+# We use DELETE c FROM t syntax to make list column c mentioned in the statement
+# (it's also important to trigger the relevant code path), but
+# also to not bother with providing condition for it in the WHERE clause.
+def test_delete_with_list_column_and_missing_clustering_key_part(cql, table2):
+    key = unique_key_string()
+    cql.execute(cql.prepare(f"INSERT INTO {table2} (id,a,b,c) VALUES ('{key}',1,1,[1])"))
```

Comment:
Again, is it important that C is a list in this test? Or it's just an arbitrary column type and would have worked the same also for an integer?

Reply:
It's important for the reasons described above. And btw, this fact is mentioned in the comments for the `test_insert_with_list_column_and_missing_clustering_key_part` which is referenced here.